pennylane/wires.py

```diff
@@ -493,6 +493,207 @@ def unique_wires(list_of_wires):
 
         return Wires(tuple(unique), _override=True)
 
+    def union(self, other):
+        """Return the union of the current Wires object and either another Wires object or an
+        iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If ``other`` is not an instance of Wires.
+
+        **Example**
+
+        >>> from pennylane.wires import Wires
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1.union(wires2)
+        Wires([1, 2, 3, 4, 5])
+        """
+        return Wires((set(self.labels) | set(_process(other))))
+
+    def __or__(self, other):
+        """Return the union of the current Wires object and either another Wires object or an
+        iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> from pennylane.wires import Wires
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1 | wires2
```

Comment:
Are such examples visible to users anywhere?

Reply:
No, don't think so. We don't want users to necessarily see the dunder methods on the docs I'm assuming, so I will move the example to somewhere visible.